src/api/graphql.ts

```diff
@@ -160,27 +177,35 @@ export async function getWalletInteractionsForContract(
         }`,
     };
 
-    const { status, ...response } = await arweave.api.post(
+    const { status, data } = await arweave.api.post<GQLResultInterface>(
       '/graphql',
       queryObject,
     );
+
     if (status !== 200) {
       throw Error(
         `Failed to fetch contracts for wallet. Status code: ${status}`,
       );
     }
 
-    if (!response.data.data?.transactions?.edges?.length) {
+    if (!data.data.transactions?.edges?.length) {
       continue;
     }
-    for (const e of response.data.data.transactions.edges) {
+
+    // remove interactions without block data
+    const validInteractions = data.data.transactions.edges.filter(
+      (i) => i.node.block && i.node.block.height && i.node.block.id,
+    );
+    // sort them using warps sort logic and adds sort keys
+    const sortedInteractions = await interactionSorter.sort(validInteractions);
```

Comment:
seems like the behavior is that if both are provided, we will first sort by block height in the GQL query and then sort by sort key via warp logic